netbox_netprod_importer/vendors/cisco/nxos.py

```diff
@@ -163,3 +165,96 @@ def _correct_and_convert_to_dict(self, cmd_output) -> dict:
                 cmd_output = re.sub('^.+\. {', '{', cmd_output, count=1)
             cmd_output = json.loads(cmd_output)
         return cmd_output
+
+    def get_interfaces_lag(self, interfaces):
+        cmd = "show port-channel summary" + self._driver_end_selection()
+        interfaces_lag = defaultdict(list)
+
+        cmd_output = self.device.cli([cmd])[cmd]
+        port_cannels = self._correct_and_convert_to_dict(cmd_output)[
+            "TABLE_channel"]["ROW_channel"]
+
+        if isinstance(port_cannels, dict):
+            port_cannels = [port_cannels]
+
+        for p in port_cannels:
+            if isinstance(p["TABLE_member"]["ROW_member"], dict):
+                member = [p["TABLE_member"]["ROW_member"]]
+            else:
+                member = p["TABLE_member"]["ROW_member"]
+            for interface in member:
+                interfaces_lag[interface["port"]] = p["port-channel"]
+
+        return interfaces_lag
+
+    def get_interface_mode(self, interface):
```

Comment:
As you seemed to have copy/paste these 2 functions (`get_interface_mode` and `get_interface_access_vlan`), maybe they could be shared by putting them in the `CiscoParser` class?

Reply:
Here I agree with you, but it is very inconvenient to catch the type of driver during the tests, and the tests need to be redone.
How to do this, I do not know.
It is necessary to determine when testing whether or not to call `self.get_abrev_if (interface)`